lib/puppet/provider/foreman_resource/rest_v3.rb

```diff
@@ -99,14 +99,16 @@ def success?(response)
   end
 
   def error_message(response)
+    fqdn = URI::parse(resource[:base_url]).host
+
     explanations = {
-      '400' => 'Something is wrong with the data sent to Foreman server',
+      '400' => "Something is wrong with the data sent to #{fqdn}",
```

Comment:
Should this then also be `sent to Foreman at #{fqdn}` ?

Reply:
I think that would be both more specific and more consistent.

I will make the change

Reply:
I'll use this consistently throughout, i.e. also changing to `"Often this is caused by invalid Oauth credentials sent to Foreman at #{fqdn}"` and `"The requested resource was not found in Foreman at #{fqdn}"` for the explanations for 401 and 404 codes